interfaces/apparmor/backend_test.go

```diff
@@ -589,38 +589,45 @@ profile "snap.samba_foo.smbd" (attach_disconnected,mediate_deleted) {
 	s.RemoveSnap(c, snapInfo)
 }
 
-// On openSUSE Tumbleweed partial apparmor support doesn't change apparmor template to classic.
-// Strict confinement template, along with snippets, are used.
-func (s *backendSuite) TestCombineSnippetsOpenSUSETumbleweed(c *C) {
-	restore := release.MockAppArmorLevel(release.PartialAppArmor)
-	defer restore()
-	restore = release.MockReleaseInfo(&release.OS{ID: "opensuse-tumbleweed"})
-	defer restore()
-	restore = osutil.MockKernelVersion("4.16.10-1-default")
-	defer restore()
-	restore = apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
-	defer restore()
-	restore = apparmor.MockIsRootWritableOverlay(func() (string, error) { return "", nil })
-	defer restore()
-	// NOTE: replace the real template with a shorter variant
-	restoreTemplate := apparmor.MockTemplate("\n" +
+func mockPartalAppArmorOnDistro(c *C, kernelVersion string, releaseID string, releaseIDLike ...string) (restore func()) {
+	restore1 := release.MockAppArmorLevel(release.PartialAppArmor)
+	restore2 := release.MockReleaseInfo(&release.OS{ID: releaseID, IDLike: releaseIDLike})
+	restore3 := osutil.MockKernelVersion(kernelVersion)
+	restore4 := apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
+	restore5 := apparmor.MockIsRootWritableOverlay(func() (string, error) { return "", nil })
+	// Replace the real template with a shorter variant that is easier to test.
+	restore6 := apparmor.MockTemplate("\n" +
 		"###VAR###\n" +
 		"###PROFILEATTACH### (attach_disconnected) {\n" +
 		"###SNIPPETS###\n" +
 		"}\n")
-	defer restoreTemplate()
-	restoreClassicTemplate := apparmor.MockClassicTemplate("\n" +
+	restore7 := apparmor.MockClassicTemplate("\n" +
 		"#classic\n" +
 		"###VAR###\n" +
 		"###PROFILEATTACH### (attach_disconnected) {\n" +
 		"###SNIPPETS###\n" +
 		"}\n")
-	defer restoreClassicTemplate()
+
+	return func() {
```

Comment:
Maybe for a future version: could we switch to testutil.BaseTest and AddCleanup() ?